content/guides/create-a-booking-payments-on-partner-side.md

```diff
@@ -0,0 +1,162 @@
+
+This document outlines the process of handling payments on the partner's side. While we offer the option to use the Smily payment gateway, this guide focuses on managing payments directly through the partner's infrastructure.
+
+## Quote creation
+
+Before proceeding with booking creation, it's necessary to confirm rental price and availability by generating a quote.
+
+~~~ruby
+require 'excon'
+
+token = "<YOUR_TOKEN>"
+api_url = "<API_URL>"
+media_type = "application/vnd.api+json"
+options = {
+  headers: {
+    "User-Agent" => "Api client",
+    "Accept" => media_type,
+    "Content-Type" => media_type,
+    "Authorization" => "Bearer #{token}"
+  }
+}
+
+request = Excon.new(URI.join(api_url, "/api/ota/v1/quotes").to_s, options)
+payload = {
+  data: {
+    attributes: {
+      "start-at": "2023-08-04",
+      "end-at": "2023-08-11",
+      "adults": 20,
+      "children": 0,
+      "rental-id": 428
+    },
+    type: "quotes"
+  }
+}
+response = request.request({
+  method: :post,
+  body: payload.to_json
+})
+
+json = JSON.parse(response.body)
+if response.status == 201
+  price = json["data"]["attributes"]["final-price"]
+  booking_url = json["data"]["attributes"]["booking-url"]
+  # Proceed to booking creation via API or redirect user to booking_url
+else
+  handle_errors(json)
+end
+~~~
+
+## Booking creation
+
+Once a successful quote is obtained, initiate a booking request by providing client details, rental information, and pricing.
+
+> Note: Ensure that the provided price matches or higher than `final-price` from the quote.
+
+~~~ruby
+token = "<YOUR_TOKEN>"
+api_url = "<API_URL>"
+media_type = "application/vnd.api+json"
+options = {
+  headers: {
+    "User-Agent" => "Api client",
+    "Accept" => media_type,
+    "Content-Type" => media_type,
+    "Authorization" => "Bearer #{token}",
+    "Idempotency-Key" => get_order_uuid # Use a unique ID for idempotency
+  }
+}
+
+request = Excon.new(URI.join(api_url, "/api/ota/v1/bookings").to_s, options)
+payload = {
+  data: {
+    attributes: {
+      "start-at": "2020-09-04T16:00:00.000Z", # "2020-09-04" works too
+      "end-at": "2020-09-11T10:00:00.000Z", # "2020-09-11" works too
+      "adults": 2,
+      "children": 1,
+      "final-price": "176.0",
+      "currency": "EUR",
+      "rental-id": 1,
+      "client-first-name": "Rich",
+      "client-last-name": "Piana",
+      "client-email": "rich@piana.com",
+      "client-phone-number": "123123123",
+      "client-country-code": "US"
+    },
+    type: "bookings"
+  }
+}
+response = request.request({
+  method: :post,
+  body: payload.to_json
+})
+
+json = JSON.parse(response.body)
+if response.status == 201
+  booking_id = json["data"]["id"]
+  # Save the booking ID for future reference
+else
+  handle_errors(json)
+end
+~~~
+
+> **Note:** If a created booking includes the not null `tentative-expires-at` field, it may be automatically canceled if no payment is made. Therefore, it's essential to proceed with payment creation.
```

Comment:
We always create a tentative booking. So booking without payment will be always canceled in 10 minutes by default (we can increase this window by request).

Reply:
Note: Acreated booking includes the not null tentative-expires-at field, it will be automatically canceled if no payment is made. Therefore, it’s essential to proceed with payment creation.